src/components/date-range/date-range.component.tsx

```diff
@@ -134,26 +221,27 @@ const DateRange = ({
   ]);
 
   const buildCustomEvent = useCallback(
-    (changedDate, newValue) => {
-      const startValue =
-        changedDate === "start" && newValue ? newValue : startDateValue;
-      const endValue =
-        changedDate === "end" && newValue ? newValue : endDateValue;
+    (
+      changedDate: string,
+      newValue: { formattedValue: string; rawValue: string | null }
```

Comment:
```suggestion
      newValue: DateInputValue
```

**suggestion**: We should do this now that we've extracted it as a separate type.